src/App.jsx

```diff
@@ -14,33 +15,116 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const ALPHABET_SORT = 'alphabet';
+const STR_LENGTH_SORT = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [goodsList, setGoodsList] = useState(goodsFromServer);
+  const [rev, setRev] = useState(false);
+  const [classActive, setClassActive] = useState('');
+  const [reverseActive, setREverseActive] = useState(true);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const getPreparedGoods = (goods, { sort, reverse }) => {
+    const preparedGoods = [...goods];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (sort) {
+      return preparedGoods.sort((good1, good2) => {
+        switch (sort) {
+          case ALPHABET_SORT:
+            return good1.localeCompare(good2);
+          case STR_LENGTH_SORT:
+            return good1.length - good2.length;
+          default:
+            return 0;
+        }
+      });
+    }
+
+    if (reverse) {
+      return preparedGoods.reverse();
+    }
```

Comment:
The reverse logic is not correctly integrated with the sorting logic. Consider applying the reverse operation after sorting if both are needed.